core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreReaderFragment.kt

```diff
@@ -0,0 +1,1508 @@
+  protected open fun configureWebViewSelectionHandler(menu: Menu?) {
+    menu?.findItem(R.id.menu_speak_text)?.setOnMenuItemClickListener {
+      tts!!.readSelection(getCurrentWebView())
+      actionMode?.finish()
+      true
+    }
+  }
+
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    bottomToolbar = bottom_toolbar
+    noOpenBookButton = go_to_library_button_no_open_book
+    toolBar = toolbar
+    contentFrame = activity_main_content_frame
+    closeAllTabsButton = tab_switcher_close_all_tabs
+    tabSwitcherRoot = activity_main_tab_switcher
+    progressBar = main_fragment_progress_view
+    activityMainRoot = activity_main_root
+    toolbarContainer = fragment_main_app_bar
+    videoView = fullscreen_video_container
+    setHasOptionsMenu(true)
+    val activity = activity as AppCompatActivity
+    WebView(activity).destroy() // Workaround for buggy webViews see #710
+    handleLocaleCheck()
+    activity.setSupportActionBar(toolbar)
+    actionBar = activity.supportActionBar
+    initHideBackToTopTimer()
+    initTabCallback()
+    toolbar.setOnTouchListener(toolbarSwipeTouchListener)
+    loadDrawerViews()
+    tableDrawerRight =
+      tableDrawerRightContainer.getHeaderView(0).findViewById(R.id.right_drawer_list)
+    addFileReader()
+    setupTabsAdapter()
+    setTableDrawerInfo()
+    setTabListener()
+    compatCallback = CompatFindActionModeCallback(activity)
+    setUpTTS()
+    setupDocumentParser()
+    loadPrefs()
+    updateTitle()
+    handleIntentExtras(activity.intent)
+    tab_switcher_recycler_view!!.adapter = tabsAdapter
+    ItemTouchHelper(tabCallback!!).attachToRecyclerView(tab_switcher_recycler_view)
+
+    // Only check intent on first start of activity. Otherwise the intents will enter infinite loops
+    // when "Don't keep activities" is on.
+    if (savedInstanceState == null) {
+      handleIntentActions(activity.intent)
+    }
+    bottom_toolbar_arrow_back.setOnClickListener { goBack() }
+    bottom_toolbar_arrow_forward.setOnClickListener { goForward() }
+    bottom_toolbar_toc.setOnClickListener { openToc() }
+    activity_main_fullscreen_button.setOnClickListener { closeFullScreen() }
+    tab_switcher_close_all_tabs.setOnClickListener { closeAllTabs() }
+    bottom_toolbar_bookmark.setOnClickListener { toggleBookmark() }
+    bottom_toolbar_bookmark.setOnLongClickListener { goToBookmarks() }
+    bottom_toolbar_home.setOnClickListener { openMainPage() }
+    activity_main_back_to_top_fab.setOnClickListener { backToTop() }
+    activity_main_button_pause_tts.setOnClickListener { pauseTts() }
+    activity_main_button_stop_tts.setOnClickListener { stopTts() }
+  }
+
+  private fun initTabCallback() {
+    tabCallback = object : ItemTouchHelper.Callback() {
+      override fun getMovementFlags(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder
+      ): Int {
+        return makeMovementFlags(
```

Comment:
I think this would look cool-er if we change it from block body to expression body